arch/x86/core/pcie.c

```diff
@@ -214,14 +219,18 @@ uint16_t pcie_msi_mdr(unsigned int irq,
 
 	if (vector == NULL) {
 		/* edge triggered */
-		return 0x4000U | Z_IRQ_TO_INTERRUPT_VECTOR(irq);
+		msg_data_reg = 0x4000U | Z_IRQ_TO_INTERRUPT_VECTOR(irq);
 	}
-
+#if !defined(CONFIG_INTEL_VTD_ICTL)
+  else{
```

Comment:
I think space after `else` is missing, also would it look better if we put `ifdef` inside `else { }`?